nodejs/awsx/ec2/vpc.ts

```diff
@@ -124,8 +126,6 @@ export class Vpc extends pulumi.ComponentResource {
             throw new Error("Cannot add InternetGateway to Vpc that already has one.");
         }
 
-        opts = { parent: this, ...opts };
-
         // See https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Internet_Gateway.html#Add_IGW_Attach_Gateway
         // for more details.
         this.internetGateway = new x.ec2.InternetGateway(name, this, args, opts);
```

Comment:
this is inside an instance helper method on top of awsx.ec2.Vpc.  With the instance methods, we pass along 'this' and 'opts' untouched.  'opts' allows the caller to choose what they want as options for this auxilliary resource.  But we also pass 'this' in case they do not specify a parent.  In that case, we will parent these auxilliary resources with 'this' resource as it can sensibly be a parent here.